### ISSUE TITLE
The epoch for every result

### ISSUE THREAD

**Issue by Ethanwl** [gjy3035/C-3-Framework#6]
How many epochs to get the corresponding results of the dataset and model? The default setting in config.py is 100 and there is no explanation. And will you release the trained models?

**Comment by gjy3035** [comment]
The current ```config.py``` dose not mean the default setting, which is only the latest setting for a specific experiments. 
If you want to reproduce the proposed results in readme, you should adopt the provided parameter setting in ```results_reports```. Each experiments setting are saved in ```setting.py``` and ```config.py```.

**Comment by Ethanwl** [comment]
When the image size is not the same in a dataset, such as SHHA. How to compute mean and std of each channel?

**Comment by gjy3035** [comment]
I don't do the experiments using CSR's setting. But I think the performance is only affected slightly. In my other experiments (not shown in this repo), I mistakenly use the other dataset's mean and std to train the model, which is close to the result of correct setting.

**Comment by Ethanwl** [comment]
![Screenshot_2019-04-02 TensorBoard](https://user-images.githubusercontent.com/31791137/55398510-4b0d1e00-557b-11e9-830a-28be68c280fb.png)
![Screenshot_2019-04-0201 TensorBoard](https://user-images.githubusercontent.com/31791137/55398530-5eb88480-557b-11e9-8e19-8bea236b7967.png)
I use your  pre-trained CSRNet model on GCC to train with UCF-QNRF for 180 epoches. The best mae 112.4 and mse 188.4 at epoch 161

**Comment by Ethanwl** [comment]
![141_170_150](https://user-images.githubusercontent.com/31791137/55398923-48f78f00-557c-11e9-9f9a-fbf61d47d745.jpg)
I use the CSRNet to predict and always get the pictures with red dot, such as the right picture.
Do you meet the same problem?

**Comment by Ethanwl** [comment]
@gjy3035  Cound you realease the pre-trained SANet model on GCC?

**Comment by gjy3035** [comment]
1. The standard CSRNet may output the negative density for unseen scenes. If you want to avoid this problem,  you would add ReLU layer on the top of CSR.
2. I will provide the download link for all pre-trained models on GCC using random splitting evaluation ASAP.

**Comment by gjy3035** [comment]
I have updated the link for each pretrained model in readme. Please visit the page to download them.

**Comment by Ethanwl** [comment]
Thank you for your pretrained model. I train the pretrained vgg_decoder model on QNRF and get the best mae 109.4 and mse 181.9 at epoch 317.
I find that the mae and mse in [https://github.com/gjy3035/C-3-Framework/blob/master/README.md](url) are not the same as [https://github.com/gjy3035/C-3-Framework/blob/master/results_reports/VGG_decoder/UCFQNRF/readme.md](url)
I have another two questions:
1. For generating density map, the kernel of window size (μ)=15 and spread parameter (σ)=4 are used in all datasets. The paper _W-Net: Reinforced U-Net for Density Map Estimation_ says this method is much better than others. Is there any internal mechanism to explain that?
2. Some paper use patch-based training and testing scheme. I think if there is no Instance Normalization in the model, this method won't improve results. Right?

**Comment by erichhhhho** [comment]
@Ethanwl Also, I noticed that in [GCC_cl/config.py line 23](https://github.com/gjy3035/C-3-Framework/blob/python3.x/results_reports/VGG_decoder/GCC_cl/config.py#L23):  `__C.NET = 'Res50' #` Which is mismatched from the txt result, and `__C.NET` should be `VGG_DECODER`. I doubted that the result and setting of VGG_Decoder may not be correct here.  

Hi @gjy3035, could you pls check if the VGG_Decoder results are all correct?

**Comment by gjy3035** [comment]
@erichhhhho 
Sorry, the ```config.py```may be mistakenly uploaded. See the [log file](https://github.com/gjy3035/C-3-Framework/blob/python3.x/results_reports/VGG_decoder/GCC_cl/01-08_00-40_GCC_VGG_DECODER_1e-05_cl.txt), the net is "VGG_DECODER".
We will upload the correct file. Before this, you can use the setting of the log file.
